mproton.go

```diff
@@ -0,0 +1,101 @@
+package mproton
+
+/*
+#cgo darwin CFLAGS: -x objective-c
+#cgo darwin CFLAGS: -Wobjc-property-no-attribute
+#cgo darwin LDFLAGS: -framework Cocoa
+#cgo darwin LDFLAGS: -framework Foundation
+#cgo darwin LDFLAGS: -framework WebKit
+
+//sudo apt-get -y install libgtk-3-dev libwebkit2gtk-4.0-dev libappindicator3-dev
+
+#cgo linux openbsd freebsd CXXFLAGS: -DWEBVIEW_GTK -std=c++11
+#cgo linux openbsd freebsd pkg-config: gtk+-3.0 webkit2gtk-4.0 appindicator3-0.1
+
+
+// sudo apt-get install -y electric-fence
+// #cgo linux LDFLAGS: -lefence
+
+#include "mproton.h"
+
+
+#include <stdio.h>
+#include <stdlib.h>
+*/
+import "C"
+import "fmt"
+
+var callbackMap = make(map[string]func(v string) (string, error))
+
+func registerCallback(name string, callback func(v string) (string, error)) int {
+	callbackMap[name] = callback
+	return 0
+}
+
+//export goTrampoline
+func goTrampoline(a int, param1 *C.char, param2 *C.char) (*C.char, *C.char) {
```

Comment:
What is the first parameter?

Reply:
Thinking of a placeholder in case functions with more parameters would be marshaled.